python/private/hermetic_runtime_repo_setup.bzl

```diff
@@ -0,0 +1,161 @@
+            ] + extra_files_glob_include,
+            # Platform-agnostic filegroup can't match on all patterns.
+            allow_empty = True,
+            exclude = [
+                "**/* *",  # Bazel does not support spaces in file names.
+                # Unused shared libraries. `python` executable and the `:libpython` target
+                # depend on `libpython{python_version}.so.1.0`.
+                "lib/libpython{}.{}.so".format(major, minor),
+                # static libraries
+                "lib/**/*.a",
+                # tests for the standard libraries.
+                "lib/python{}.{}/**/test/**".format(major, minor),
+                "lib/python{}.{}/**/tests/**".format(major, minor),
+                "**/__pycache__/*.pyc.*",  # During pyc creation, temp files named *.pyc.NNN are created
+            ] + extra_files_glob_exclude,
+        ),
+    )
+    cc_import(
+        name = "interface",
+        interface_library = "libs/python{}{}.lib".format(major, minor),
+        system_provided = True,
+    )
+
+    native.filegroup(
+        name = "includes",
+        srcs = native.glob(["include/**/*.h"]),
+    )
+    cc_library(
+        name = "python_headers",
+        deps = select({
+            "@bazel_tools//src/conditions:windows": [":interface"],
+            "//conditions:default": None,
+        }),
+        hdrs = [":includes"],
+        includes = [
+            "include",
+            "include/python{}.{}".format(major, minor),
+            "include/python{}.{}m".format(major, minor),
+        ],
+    )
+    cc_library(
+        name = "libpython",
+        hdrs = [":includes"],
+        srcs = select({
+            "@platforms//os:linux": [
+                "lib/libpython{}.{}.so".format(major, minor),
+                "lib/libpython{}.{}.so.1.0".format(major, minor),
+            ],
+            "@platforms//os:macos": ["lib/libpython{}.{}.dylib".format(major, minor)],
+            "@platforms//os:windows": ["python3.dll", "libs/python{}{}.lib".format(major, minor)],
+        }),
+    )
+
+    native.exports_files(["python", python_bin])
+
+    # Used to only download coverage toolchain when the coverage is collected by
+    # bazel.
+    native.config_setting(
+        name = "coverage_enabled",
+        values = {"collect_code_coverage": "true"},
+        visibility = ["//visibility:private"],
+    )
+    py_runtime_kwargs = {}
+    if coverage_tool:
+        # Earlier versions of Bazel don't support this attribute.
+        py_runtime_kwargs["coverage_tool"] = coverage_tool
+
+    py_runtime(
+        name = "py3_runtime",
+        files = [":files"],
+        interpreter = python_bin,
+        interpreter_version_info = {
+            "major": str(major),
+            "micro": str(python_version_info[2]),
+            "minor": str(minor),
+        },
+        python_version = "PY3",
+        implementation_name = "cpython",
+        pyc_tag = "cpython-{}{}".format(major, minor),
```

Comment:
I am so used to this being something like `cp{major}{minor}` in the wheels. It's interesting that this is different, should we have a small comment here linking to some docs?

Reply:
Linked to the pep